### ISSUE TITLE
Make femtolisp a submodule

### ISSUE THREAD

**Issue by chigoncalves** [FemtoEmacs/Femto-Emacs#3]
It makes sense to have [JeffBezanson/femtolisp](https://github.com/JeffBezanson/femtolisp) as a _submodule_ of
`FemtoEmacs`, since it is a third party project, it makes lot easier
to integrate _upstream_ changes into `FemtoEmacs`, I suggest
incorparate the current changes made to `FemtoEmacs/femtolisp`
into your own fork of [JeffBezanson/femtolisp](https://github.com/JeffBezanson/femtolisp), and then have it as the 
_submodule_ of `FemtoEmacs`.

I suggest this approach:
- Create a new _branch_ in [FemtoEmacs/femtolisp](https://github.com/FemtoEmacs/femtolisp), this branch can be
  named `femtoo-emacs` for instance.
- Incorporate current changes from `FemtoEmacs/femtolisp` into 
  the _branch_.
- Delete `FemtoEmacs/femtolisp` and then add your own _fork_ as 
  a `FemtoEmacs` _submodule_.

Chi

**Comment by hughbarney** [comment]
I agree.   I would like to get rid of this altogther and have the makefile pull the femtolisp code direct from Jeff repository and ensure any of our changes were put into seperate files etc.   Then have a make target called femtolisp which grabbed, built it and applied our changes.   I'm not that familiar with our changes to femtolisp as lucas did the work here.  But I am going to go through them in detail at some point. 

Also would like any changes / improvements we have made to femtolisp to be be a pull request to Jeff's codebase. Eg the maths code needs to go into its own math.c file and push to femtolisp.

BTW I will invite you to the FemtoEmacsDevs chat room on gitter.
Hugh

**Comment by chigoncalves** [comment]
> I would like to get rid of this altogther and have the makefile pull the femtolisp code direct from Jeff repository and ensure any of our changes were put into seperate files etc.

I would recommend  keeping your own _fork_ of Jeff's repository, cases where Jeff's refuses to _merge_
changes from your _fork_.

I'm already trying to "merge" addins from `Femto-Emacs/femtolisp` to my _fork_ of _femtolisp_.

Chi

**Comment by lassik** [comment]
https://github.com/lambdaconservatory/femtolisp is soliciting patches from FemtoLisp users. I'm happy to give write access to anyone who wants it.

**Comment by hughbarney** [comment]
Update:

I now manage the Femto repository at:
https://github.com/hughbarney/femto/blob/master/README.md

I will take future updates on that repository.

I abandonned using FemtoLisp and use a tinyLisp implementation that is part of the actual codebase.  The code was public domain and I was free to incorporate it fully into Femto.  I also extended the TinyLisp to support strings.  The result is very effective and stable.  Plus we dont have to compile 12K of FemtoLisp with 2K of Editor code lines.
